components/core/src/clp/ffi/ir_stream/ir_unit_deserialization_methods.cpp

```diff
@@ -460,6 +461,24 @@ auto deserialize_value_and_construct_node_id_value_pairs(
 }
 }  // namespace
 
+auto get_ir_unit_type_from_tag(encoded_tag_t tag) -> IrUnitType {
+    // First, we check the tags that have one-to-one IR unit mapping
+    if (cProtocol::Eof == tag) {
+        return IrUnitType::EndOfStream;
+    }
+    if (cProtocol::Payload::UtcOffsetChange == tag) {
+        return IrUnitType::UtcOffsetChange;
+    }
+
+    // Then, check tags that may match any byte within a continuous range
+    if ((tag & cProtocol::Payload::SchemaTreeNodeMask) == cProtocol::Payload::SchemaTreeNodeMask) {
+        return IrUnitType::SchemaTreeNodeInsertion;
+    }
+
+    // If not match, it is assumed to be a log event
+    return IrUnitType::LogEvent;
```

Comment:
Can we assume that a tag is never invalid? As far as I can tell, `deserialize_tag` will succeed for any numeric value, so it isn't obvious that we can actually trust that a tag can't be corrupt/invalid.

If it is actually fine to treat any remaining numeric value as a log event and not fail here we should document why imo.
Although, I feel ideally we validate the tag is a log event and return a result to handle an invalid tag (or maybe an invalid `IrUnitType` value).

Reply:
Sure. I think we can validate tags here

Reply:
added